bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/protocol/EmbyClientSocket.java

```diff
@@ -0,0 +1,209 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.emby.internal.protocol;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.emby.internal.model.EmbyPlayStateModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * EmbyClientSocket implements the low level communication to Emby through
+ * websocket. Usually this communication is done through port 9090
+ *
+ * @author Zachary Christiansen - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class EmbyClientSocket {
+
+    private static final int REQUEST_TIMEOUT_MS = 10000;
+    private final Logger logger = LoggerFactory.getLogger(EmbyClientSocket.class);
+    private final ScheduledExecutorService scheduler;
+    private final Gson mapper = new Gson();
+    private final EmbyClientSocketEventListener eventHandler;
+    private final WebSocketClient client;
+    private final ClientUpgradeRequest request = new ClientUpgradeRequest();
+    private final EmbyWebSocketListener socket = new EmbyWebSocketListener();
+
+    private @Nullable URI uri;
+    private @Nullable Session session;
+    private int bufferSize;
+
+    private volatile boolean shouldReconnect = true;
+    private int reconnectAttempts = 0;
+
+    public EmbyClientSocket(EmbyClientSocketEventListener handler, @Nullable URI setUri,
+            ScheduledExecutorService setScheduler, int setBufferSize, WebSocketClient sharedWebSocketClient) {
+        this.eventHandler = handler;
+        this.uri = setUri;
+        this.scheduler = setScheduler;
+        this.bufferSize = setBufferSize;
+        this.client = sharedWebSocketClient;
+    }
+
+    public synchronized void open() throws Exception {
+        if (isConnected()) {
+            logger.debug("already open");
+            return;
+        }
+        if (!client.isStarted()) {
+            client.setMaxTextMessageBufferSize(bufferSize);
+        }
```

Comment:
1. The client will always be started here.
2. I don't think you should set this as this is globally for all bindings.

if this is really needed, you should not use the shared client.

Reply:
removed it.  Which also meant that config parameters for setting the buffer size are no longer needed so those have been removed as well.  

i think this was a hold over from my early testing with developing the binding where i thought that the buffer size needed to be larger when in fact it does not need to be.